### ISSUE TITLE
Versioning / protocol metadata serving (discussion and/or spec)

### ISSUE THREAD

**Issue by tkstanczak** [ethereum-oasis/eth1.x-JSON-RPC-API-standard#10]

**Comment by MicahZoltu** [comment]
Assuming we assert that we won't do any backward incompatible changes, what is the advantage of introducing versioning?

**Comment by alcuadrado** [comment]
I'd say is mostly communicational. Implementers can state that they support up to X version.

Having said that, how can we deal with version numbers when the EIP process is pretty decentralized. We won't have a clear sequence of EIPs to derive the version numbers from, as multiple can be pushed in parallel. 

Maybe this leads to a bigger question: will the spec evolve through EIPs?

**Comment by tkstanczak** [comment]
Suggestion from the call: OpenRPC spec can be returned by the JSON RPC itself.
rpc.discover

**Comment by MicahZoltu** [comment]
I see.  So as we add methods to the spec, we would bump the version number and clients can say they are "spec compliant up to version X".  Do we version each method so that someone can be spec compliant for version X of `eth_sendTransaction` while they may not be spec compliant for any version of `eth_call`?

**Comment by tkstanczak** [comment]
Yes, we can go:
RPC level
Module level
Method level

What is everybody thinking?

**Comment by MicahZoltu** [comment]
If we go down this path (I'm still not a huge fan, but I'm not as opposed as originally), then I think method level versioning is the only thing that really makes sense to me.  The other two feel like they will result in significantly less utility since it is very likely that a given client will implement 90% of a version but not 100% at which point you are back to making statements like "I implement spec v1 for the following methods" or "I implement spec v1 expect for the following methods".  Makes more sense to me to just be able to assert compatibility with each.

The reason I'm still not convinced that this is beneficial/useful is that it isn't clear to me that there will ever be more than one version.  We have "pre-spec" endpoints, and we'll have "to spec" endpoints but I don't foresee a path for us having `eth_sendTransaction v1` and `eth_sendTransaction v2`.  What would that look like exactly?
